include/SampleBuffer.h

```diff
@@ -269,6 +270,9 @@ class LMMS_EXPORT SampleBuffer : public QObject, public sharedObject
 	}
 
 
+	sample_t **m_userAntiAliasWaveTable;
```

Comment:
Which type would be suitable for holding the wave table? Maybe `std::vector<std::vector<sample_t>>` or some `std::unique_ptr`?

Reply:
If I'm not mistaken, the number of wave tables and the wave table length is given by a constant? So perhaps even `std::array`? I will investigate.

Reply:
> So perhaps even `std::array`?

Looks like a valid option. We can even wrap this with `std::unique_ptr` for use in `SampleBuffer`s if desired.
```cpp
using wavetable_t = std::array<std::array<sample_t>, WAVETABLE_LENGTH>, WAVE_TABLES_PER_WAVEFORM_COUNT>;
std::unique_ptr<wavetable_t> m_userAntiAliasWaveTable;
````

Reply:
So, I tried replacing it with the `unique_ptr<std::array<std:array<sample_t>>>` and while it "works", there is some serious performance problem. I tried to illustrate with a video, but GitHub is being difficult and won't let me upload it even in an archive; basically triple osc. fills up the CPU graph and renders about 1 or 2 buffers worth of data every second.

Because `Oscillator::wtSample()` needs the array in a raw format that is hard to supply from the calling side (I would have to somehow serialize all the `.data()` from the `std::arrays` to resemble the original form), I made an overloaded version that does the same as the original, but takes the 2D `std::array` instead. AFAIK that's the only thing on the "critical path", so it must be the cause. But I can't see how -- `std::array` is supposed to be a very thin wrapper, but the performance I see is worse than Java app running on a washing machine.